protocol/messenger_edit_message_test.go

```diff
@@ -90,6 +90,77 @@ func (s *MessengerEditMessageSuite) TestEditMessage() {
 	s.Require().Equal(ErrInvalidEditOrDeleteAuthor, err)
 }
 
+func (s *MessengerEditMessageSuite) TestEditEmojiMessage() {
+	theirMessenger := s.newMessenger()
+	defer TearDownMessenger(&s.Suite, theirMessenger)
+
+	theirChat := CreateOneToOneChat("Their 1TO1", &s.privateKey.PublicKey, s.m.transport)
+	err := theirMessenger.SaveChat(theirChat)
+	s.Require().NoError(err)
+
+	ourChat := CreateOneToOneChat("Our 1TO1", &theirMessenger.identity.PublicKey, s.m.transport)
+	err = s.m.SaveChat(ourChat)
+	s.Require().NoError(err)
+
+	inputMessage := buildTestEmojiMessage(*theirChat)
+
+	sendResponse, err := theirMessenger.SendChatMessage(context.Background(), inputMessage)
+	s.NoError(err)
+	s.Require().Len(sendResponse.Messages(), 1)
+
+	response, err := WaitOnMessengerResponse(
+		s.m,
+		func(r *MessengerResponse) bool { return len(r.messages) > 0 },
+		"no messages",
+	)
+	s.Require().NoError(err)
+	s.Require().Len(response.Chats(), 1)
+	s.Require().Len(response.Messages(), 1)
+
+	ogMessage := sendResponse.Messages()[0]
+
+	messageID, err := types.DecodeHex(ogMessage.ID)
+	s.Require().NoError(err)
+
+	editedText := "edited text"
+	editedMessage := &requests.EditMessage{
+		ID:          messageID,
+		Text:        editedText,
+		ContentType: protobuf.ChatMessage_TEXT_PLAIN,
```

Comment:
Here I think `ContentType` should be `ChatMessage_EMOJI`, otherwise we're not exercising the new logic from this PR.

Reply:
@ilmotta initially the content type of the message we sent is `EMOJI` and after editing the message the content type is `TEXT_PLAIN` and we want to test that after eddting the emoji message the content type is also updated to text. The test will not pass if in this case the contentType is `ChatMessage_EMOJI`

Reply:
@jo-mut, if I remove all changes in this PR from the function `applyEditMessage`, the tests still pass, so the test is not testing your new implementation.